dockstore-webservice/src/main/java/io/dockstore/webservice/resources/CollectionHelper.java

```diff
@@ -112,7 +113,9 @@ public void evictAndAddEntries(Collection collection) {
             }
         });
         collection.setCollectionEntries(collectionEntries);
+        List<CollectionEntry> collectionNotebooks = entryDAO.getCollectionNotebooks(collection.getId());
         collection.setWorkflowsLength(collectionWorkflows.size() + (long)collectionWorkflowsWithVersions.size());
```

Comment:
Here the workflows length will include notebooks, because `getCollectionWorkflows` returns notebooks and workflows. So if you have 2 notebooks and 2 bio workflows, the response will include:

* Workflows length of 4
* Notebooks length of 2

But I would expect

* Workflows length of 2
* Notebooks length of 2

Reply:
Could use some tests as well before we go refactoring taking into account @svonworl 's suggestion